Description: ip_mreq_source is defined in all glibc not just on kfreebsd.
 Fix hurd FTBFS

--- a/groupsock/GroupsockHelper.cpp
+++ b/groupsock/GroupsockHelper.cpp
@@ -457,12 +457,14 @@
 // commands, and a special structure (ip_mreq_source).  If the include files
 // didn't define these, we do so here:
 #if !defined(IP_ADD_SOURCE_MEMBERSHIP)
+#ifndef __GLIBC__
 struct ip_mreq_source {
   struct  in_addr imr_multiaddr;  /* IP multicast address of group */
   struct  in_addr imr_sourceaddr; /* IP address of source */
   struct  in_addr imr_interface;  /* local IP address of interface */
 };
 #endif
+#endif
 
 #ifndef IP_ADD_SOURCE_MEMBERSHIP
 
